app/src/main/java/deakin/gopher/guardian/view/general/LoginActivity.kt

```diff
@@ -19,16 +28,25 @@ import deakin.gopher.guardian.services.EmailPasswordAuthService
 import deakin.gopher.guardian.services.NavigationService
 
 class LoginActivity : AppCompatActivity() {
+    private lateinit var gsoClient: GoogleSignInClient
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_login)
         val mEmail: EditText = findViewById(R.id.Email)
         val mPassword: EditText = findViewById(R.id.password)
         val progressBar: ProgressBar = findViewById(R.id.progressBar)
         val loginButton: Button = findViewById(R.id.loginBtn)
+        val loginGoogleButton: Button = findViewById(R.id.loginGoogleBtn)
```

Comment:
Cast to a `SignInButton` from `com.google.android.gms.common.SignInButton` for the standard button type

Reply:
Yeah crashes without this apparently. I would have thought Button would be the superclass of SignInButton and the upcasting would be handled automatically... apparently not!